engine/src/test/java/org/hibernate/validator/test/constraintvalidator/ConstraintValidatorPayloadTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.test.constraintvalidator;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static org.hibernate.validator.testutil.ConstraintViolationAssert.assertNoViolations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import javax.validation.Constraint;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.Payload;
+import javax.validation.Validator;
+import javax.validation.metadata.ConstraintDescriptor;
+
+import org.hibernate.validator.HibernateValidatorConfiguration;
+import org.hibernate.validator.HibernateValidatorFactory;
+import org.hibernate.validator.cfg.ConstraintMapping;
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidator;
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorInitializationContext;
+import org.hibernate.validator.testutil.TestForIssue;
+import org.hibernate.validator.testutils.ValidatorUtil;
+import org.testng.annotations.Test;
+
+/**
+ * @author Matthias Kurz
+ */
+@TestForIssue(jiraKey = "HV-1529")
+public class ConstraintValidatorPayloadTest {
```

Comment:
Generally, it looks good. The only issue is that we don't check at all that the validator is validating anything: I mean if the constraint was not exercised at all, you would have the exact same result.

It would be nice if you could add failure cases too so that we are sure the validator is working.

Testing with another type in the entity and checking that the violation returned is a `PayloadConstraint` constraint would be nice. You  can use this:
```
import static org.hibernate.validator.testutil.ConstraintViolationAssert.assertThat;
import static org.hibernate.validator.testutil.ConstraintViolationAssert.violationOf;

[...]

assertThat( violations ).containsOnlyViolations(
		violationOf( PayloadConstraint.class )
);
```